### PR TITLE
Add -Wunused-packages to common warnings

### PR DESCRIPTION
**Pull request by jhrcek**

### CODE REVIEW

File: haskell-language-server.cabal [hunk 1]
```diff
@@ -42,18 +42,13 @@ common defaults
 common test-defaults
   ghc-options:        -threaded -rtsopts -with-rtsopts=-N
 
-common common-deps
```
Comment (jhrcek) [inline haskell-language-server.cabal:45]: Deleting this for now as it's contributing more to unused-packages warnings than it's saving duplication..
Will reintroduce this later where it makes sense.

File: plugins/hls-semantic-tokens-plugin/test/Main.hs [hunk 2]
```diff
@@ -194,11 +193,9 @@ semanticTokensTests =
       goldenWithSemanticTokensWithDefaultConfig "type family" "TTypefamily",
       goldenWithSemanticTokensWithDefaultConfig "TUnicodeSyntax" "TUnicodeSyntax",
       goldenWithSemanticTokensWithDefaultConfig "TQualifiedName" "TQualifiedName"
-      -- it is not supported in ghc92
-#if MIN_VERSION_ghc(9,4,0)
-      , goldenWithSemanticTokensWithDefaultConfig "TDoc" "TDoc"
-#endif
     ]
+      -- not supported in ghc92
```
Comment (jhrcek) [inline plugins/hls-semantic-tokens-plugin/test/Main.hs:197]: CC @soulomoon I used this value-level CPP-less way to ensure this test only runs with GHC 9.4 and newer.
Did this to be able to get rid of "unused ghc dependency" warning.

File: haskell-language-server.cabal (hunk 1 above)
Comment (michaelpj) [reply haskell-language-server.cabal:45]: mostly a workaround for not yet having https://github.com/haskell/cabal/issues/9569

File: plugins/hls-semantic-tokens-plugin/test/Main.hs (hunk 2 above)
Comment (soulomoon) [reply plugins/hls-semantic-tokens-plugin/test/Main.hs:197]: Thanx, it looks good

File: ghcide/ghcide.cabal [hunk 3]
```diff
@@ -283,13 +280,6 @@ executable ghcide
   if !flag(executable)
     buildable: False
 
-  if flag(ekg)
```
Comment (jhrcek) [inline ghcide/ghcide.cabal:286]: ekg dep is only direct dep of ghcide library, not of the executable.

**Comment by jhrcek** [comment]
Ready for review. I'm quite happy about the resulting cleanup :hugs:

**Review by soulomoon — APPROVED** [review]
LGTM！

**Review by michaelpj — APPROVED** [review]
Excellent

**Comment by michaelpj** [comment]
Looks like a genuine failure, @soulomoon introducing some new warnings for you ;)

**Comment by jhrcek** [comment]
These look more like flaky tests to me. The unused-packages above the failure shouldn't fail the build per-se :thinking: 
Trying one more rerun..

**Comment by soulomoon** [comment]
oops, sorry man, I might not have this opportunities next time with this pr merged .😂

**Comment by jhrcek** [comment]
No problem, just the usual birth pains of trying to enforce stricter checks :smile:

**Pull request merged**